statuslight_control/lib/statuslight_control/statuslight.ex

```diff
@@ -1,26 +1,38 @@
 defmodule StatuslightControl.Statuslight do
   use GenServer
+  @renderer Application.get_env(:statuslight_control, :renderer)
+  @pixel_width 12
+  @pixel_height 1
 
-  def start_link(renderer) do
-    GenServer.start_link(__MODULE__, renderer, name: StatusLight)
+  def start_link(_) do
+    GenServer.start_link(__MODULE__, %{color: "green"}, name: __MODULE__)
   end
 
   def stop do
-    GenServer.stop(StatusLight)
+    GenServer.stop(__MODULE__)
   end
 
-  def init(renderer) do
-    {:ok, %{renderer: renderer} }
+  @impl true
+  def init(state) do
+    {:ok, state}
   end
 
   def set_color(color) do
-    GenServer.cast(StatusLight, {:set_color, color})
+    GenServer.cast(__MODULE__, {:set_color, color})
   end
 
-  def handle_cast({:set_color, color}, state) do
+  def current_color() do
+    GenServer.call(__MODULE__, :get_color)
+  end
+
+  @impl true
+  def handle_call(:get_color, _from, %{color: color} = state), do: {:reply, color, state}
+
+  @impl true
+  def handle_cast({:set_color, color}, _state) do
     bc_color = Blinkchain.Color.parse(color)
-    state.renderer.fill(%Blinkchain.Point{x: 0, y: 0}, 12, 1, bc_color)
-    state.renderer.render()
-    {:noreply, state}
+    @renderer.fill(%Blinkchain.Point{x: 0, y: 0}, @pixel_width, @pixel_height, bc_color)
+    @renderer.render()
```

Comment:
This should still work, the blinkchain lib has its own Genserver it's storing state in for these things.